### PR TITLE
InverseKinematicsStudy only converts purely Rotational coordinates to degrees.

### PR DESCRIPTION
**Pull request by aseth1**
Fixes issue  #2470

### Brief summary of changes
- Explicitly check for Rotational coordinates (instead of not Translational)

### Testing I've completed
- reran opensense IK and loaded calibrated model and resultant motion in OpenSim GUI where patellas moved as expected

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2473)

### CODE REVIEW

File: OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp [hunk 1]
```diff
@@ -113,18 +113,21 @@ runInverseKinematicsWithOrientationsFromFile(Model& model,
...
             coord.getOutput("value"), coord.getName());
-        if (coord.getMotionType() == Coordinate::Translational) {
-            transCoordNames.push_back(coord.getName());
-            coord.set_locked(true);
+        if (coord.getMotionType() == Coordinate::Rotational) {
+            rotCoordNames.push_back(coord.getName());
+        }
+        else if(coord.getMotionType() == Coordinate::Translational) {
+            coord.setDefaultLocked(true);
         }
```
Comment (aymanhab) [inline OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: Is there difference in behavior between set_locked and setDefaultLocked?

File: OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp [hunk 2]
```diff
@@ -195,11 +198,11 @@ runInverseKinematicsWithOrientationsFromFile(Model& model,
...
+    // but only for rotational coordinates
     for (size_t i = 0; i < report.getNumColumns(); ++i) {
-        auto it = find( transCoordNames.begin(), transCoordNames.end(),
+        auto it = find(rotCoordNames.begin(), rotCoordNames.end(),
                         report.getColumnLabel(i) );
-        if (it == transCoordNames.end()) { // not found = not translational
+        if (it != rotCoordNames.end()) { // found a rotational coordinate
             auto repVec = report.updDependentColumnAtIndex(i);
             repVec *= SimTK::Real(SimTK_RTD);
         }
```
Comment (aymanhab) [inline OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: Any reason not to use the method convertRadianToDegrees e to have these checks and conversions done in one pace?

File: OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp (hunk 1 above)
Comment (aseth1) [reply OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: No difference. `setDefaultLocked` is more explicit about the default state of the coordinate is locked.

File: OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp (hunk 2 above)
Comment (aseth1) [reply OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: I didn't want to convert to a `Storage`, but I see that Shrinidhi added a `TimeSeriesTable` version. I'll try that.

Comment (aseth1) [reply OpenSim/Simulation/OpenSense/InverseKinematicsStudy.cpp]: I remember now,  conversion fails if the "inDegrees" key is not specified. I've fixed that now too.

**Comment by jimmyDunne** [comment]
I built this branch and re-ran the tracking, but I still seem to be getting the same issue. I have attached the model and the output kinematics. 

[opensense_patella.zip](https://github.com/opensim-org/opensim-core/files/3209344/opensense_patella.zip)

**Comment by aseth1** [comment]
@aymanhab thanks for the suggestion. Looks a lot cleaner and nice to centralize the conversion too. Should be good to go.

**Comment by aseth1** [comment]
@jimmyDunne thanks for reviewing and testing.

**Pull request merged**